src/node.h

```diff
@@ -617,6 +618,33 @@ NODE_EXTERN v8::MaybeLocal<v8::Value> PrepareStackTraceCallback(
     v8::Local<v8::Value> exception,
     v8::Local<v8::Array> trace);
 
+// Writes a diagnostic report to a file. If filename is not provided, the
+// default filename includes the date, time, PID, and a sequence number.
+// The report's JavaScript stack trace is taken from err, if present.
+// If isolate is nullptr, no information about the JavaScript environment
+// is included in the report.
+// Returns the filename of the written report.
+NODE_EXTERN std::string TriggerNodeReport(v8::Isolate* isolate,
```

Comment:
thinking aloud here: assuming APMS embed node, are there any restrictions on the invocation sequence or context? for example, i) this can be invoked only after node is fully initialised (may not be obvious to embedders?) ii) this cannot be invoked from another thread (assuming the embedder has non-node threads). If so, is this worth mentioning here?

Reply:
Thanks for the idea. I believe we don't have any particular restrictions on these APIs regarding the current public embedder/add-on APIs:

i) if the `Environment` is created with `node::CreateEnvironment`, the returned `Environment` should already be bootstrapped and it is safe to call into JavaScript. Is there any specific case you believe we should take care of?

ii) IIUC most APIs in `node.h` are not threadsafe unless they are explicitly marked as threadsafe. So I don't think it is necessary to document thread safety for these particular methods.

Reply:
> Is there any specific case you believe we should take care of?

no, I wasn't aware of `node::CreateEnvironment` leads to a fully initialised node. thanks for clarification.